gridstatus/isone.py

```diff
@@ -406,40 +408,37 @@ def get_lmp(
 
         now = pd.Timestamp.now(tz=self.default_timezone)
         if market == Markets.REAL_TIME_5_MIN:
-            # todo handle intervals for current day
-            intervals = ["00-04", "04-08", "08-12", "12-16", "16-20", "20-24"]
+            intervals = self.lmp_real_time_intervals[:]
 
-            # optimze for current day
             if now.date() == date.date():
-                hour = now.hour
-                # select completed 4 hour intervals based on current hour
-                intervals = intervals[: math.ceil((hour + 1) / 4) - 1]
+                intervals = self._filter_intervals_in_range(
```

Comment:
i sort of prefer the current logic that converts the requested time into the index of the interval array rather than take the names of the intervals and converting them to datetime objects.

i think the logic would be something like, which strikes me as a lot easier to maintain. i didn't run this so there could be off by one errors. 

```python
# converts hour into the index 
# in 4 hour interval array
def _hour_to_interval_index(hour):
    if not 0 <= hour <= 23:
        raise ValueError("Hour must be between 0 and 23")
    return hour // 4

# no need to get data for future intervals
last_interval_hour = min(now.hour, end.hour) if end else now.hour
intervals = intervals[_hour_to_interval_index(date.hour) : _hour_to_interval_index(last_interval_hour)+1]
```

Reply:
you might need to change my logic to only include the completed intervals. I think that is why we didn't previously need the `try...except` you added

Reply:
Do you know how long after the interval is completed the files are available? I think we may want to keep the `try...except`.

Reply:
i dont not. im find with that approach

Reply:
Updated to use your simpler logic. The `try` has been kept.